### ISSUE TITLE
Looking for a new maintainer

### ISSUE THREAD

**Issue by gnunn1** [gnunn1/tilix#1700]
As many have probably noticed, activity on tilix has dwindled over the last year or so. From my perspective it does everything I need in a terminal emulator (well more actually) and I have little interest in adding features or even fixing bugs in features I don't use. I also want to spend more time on things more closely related to my day job and unfortunately that's not terminal emulators, GUIs, D or GNOME.

As a result I think some new blood with respect to Tilix would be beneficial. If you are interested in taking over the project let me know, ideally it would be someone who is already involved but we can work out a process for someone new.

Now does this mean tilix is completely dead if no one takes over? Probably not simply because I still use it everyday and will likely continue to put in the minimum effort to keep it working. However my motivation to push formal releases out the door versus just updating code in the repo will likely be pretty low. 

With new energy I think there are some interesting directions Tilix could go so hopefully someone will be up to taking it on.

**Comment by auvipy** [comment]
I am a Python and C programmer and a regular Linux user. You could consider me as a potential maintainer if you feel so.

**Comment by Crestwave** [comment]
@auvipy Tilix is written in D, though.

**Comment by auvipy** [comment]
Yes, I saw the code, and it seems very familiar with C with more cleaner syntax!!

**Comment by andrei-pavel** [comment]
I write C++ on my day-to-day job and I also have prior Go experience. However, I have the following philosophy regarding programming languages on new native projects that I would ever start:
- For operating systems, drivers, kernel, embedded projects, use Rust for safety
- For high performance computing, use C++
- If you want to leverage a great concurrency runtime for something like blockchain, use Go
- For everything else, use D

D is a well polished natively compiled language which makes it my safety net default language. I do not have practical experience outside a minimalistic boilerplate[1], but I have been watching it's updates on a regular basis.

I love tilix and have been using it for about a year and would be humbled and diligent in maintaining it.

I have contributed to open-source before, but not to tilix and before any consideration for me is had, I should at least try to fix a few issues or add a few features which I will start to look at today in my spare time. If you have a process set in place, let me know, email is in the profile description.

[1] https://github.com/andrei-pavel/d-boilerplate

**Comment by phw** [comment]
Just wanted to comment here because I have contributed some patches to Tilix since it's early days. I use Tilix daily, and I will for sure continue submitting patches if something itches me. In particular I will look into the localization of Tilix and make sure this stays up-to-date. I also will continue keeping an eye on the issues and commenting there if it is something I know about.

What I can not offer right now is taking over parts of real maintainership. I am pretty much in a similar situation as @gnunn1, I am already struggling to maintain some of my own projects properly.

Thanks to @gnunn1 for this excellent piece of software. It has been a joy using it in the last 3 years. I hope somebody will take over maintaining it.

**Comment by gnunn1** [comment]
@auvipy @andrei-pavel I don't have a set process in place, I think the easiest way as @andrei-pavel suggested is to start looking at issues and submitting PRs, comments to the issues. If things work out over the course of a month I'll add you to the repo so you can start submitting things directly.

If you need pointers to specific issues let me know and I'm happy to point out some that are smaller tasks.

I would also be happy to schedule a remote call where I'm happy to do a walkthrough of the code and project structure as well, answer questions, etc if it would be useful.

**Comment by ximion** [comment]
Have you thought about moving the project under the GNOME umbrella directly and have it maintained there? (You'd still need a new maintainer, but you would also have GNOME-knowledgeable people taking care of the basic project needs).

**Comment by gnunn1** [comment]
@ximion I have had discussions with some folks at GNOME about this in the past and decided against it for various reasons. I'd prefer letting a new maintainer make that discussion then make it for them.

**Comment by dogweather** [comment]
👍 for choosing multiple good candidates.

**Comment by ximion** [comment]
@gnunn1 That sounds like a very good plan :-)

**Comment by ernestask** [comment]
> @ximion I have had discussions with some folks at GNOME about this in the past and decided against it for various reasons. I'd prefer letting a new maintainer make that discussion then make it for them.

How far in the past are we talking here? We now have a World namespace for community projects.

**Comment by gnunn1** [comment]
@ernestask It was at the time of the migration to GitLab, however my point remains, I'm not making any changes on this front as I prefer to leave the decision to the next maintainer.

**Comment by auvipy** [comment]
Hi, I recently joined the python packaging authority to maintain a few official packages. So my foss time will be invested there. I am no longer in a position to consider myself as a prospective maintainer of tilix anymore.

**Comment by Mutnick** [comment]
@gunn1 Whatever happens -- thanks for Tilix! 

Oh, A little issue maybe not worth an opening an issue... Updating today I noticed the constants.d version # wasn't bumped up, and still lists as version 1.9.1.

Thanks again

**Comment by kriansa** [comment]
@gnunn1 I apologize for using this thread for this purpose, but I would like to thank you very much for all your work on Tilix. 👏 

I too find the current features just enough for what I need and it serves me perfectly.

**Comment by russel** [comment]
@gnunn1 I use Tilix all the time since it has the one feature that got removed from Terminal, transparency. I don't use many of the features of Tilix, it already provides everything I need. Thus for me bug fix merge requests are all that Tilix needs – though I have never really studied the code or made a merge request myself. Might a way forward be to create an organisation with yourself and a couple of other people as owners/admins either on GitHub or GitLab (*) and just letting things happen. A project is defined more by its usage than by its evolution, especially when forced. Tilix is alive and well because it is used, not because of the rate of change of features.

(*) Personally I am beginning to prefer GitLab over GitHub principally due to its CI integration. GitHub though still has much more mindshare than GitLab. I have been tempted in other projects to use GitLab as the primary place for a project but to have a read-only mirror on GitHub with a pointer to the GitLab project.

**Comment by teresaejunior** [comment]
This seems interesting: https://www.codeshelter.co/
>Code Shelter is a collective of volunteer software developers that aims to help with maintaining popular open source projects whose authors need a hand or don't have the time to maintain them any more.

**Comment by gnunn1** [comment]
Unfortunately no one has stepped up to take over maintainership and issues are piling up. I'm considering putting a deprecated notice on the front page of this repo and starting the process of having tilix removed from the various linux distro repos as unmaintained software. I don't think it's fair to users who are unaware of the status of tilix unless they happen to drill into this issue.

Thoughts?

**Comment by ximion** [comment]
I think before doing something radical like that, moving it to a place like the GNOME community area in their Gitlab and having it team-maintained there may be better.
There seems to be enough interest in Tilix to make this work, and as long as you have that it basically only needs someone to make releases (which is desperately needed, getting the old release to build in Debian was a struggle without taking loads of patches from Git master).

**Comment by gnunn1** [comment]
I don't think moving unmaintained software to GNOME where it remains unmaintained is going to help. GNOME folks are not familiar with D and as far as I can tell there is no team maintenance, each project is managed by it's own set of maintainers.

**Comment by russel** [comment]
@gnunn1, @ximion didn't say "unmaintained", he said "team maintained". To kill off Tilix, which has a very great advantage over GNOME Terminal for many people, because no one person feels able to step up seems to be the anti-thesis of FOSS. To pass on maintainership to a team seems like a good future for Tilix. It really should be owned by an organisation rather than an individual exactly to have a team provide the continuity that an individual cannot provide.

**Comment by gnunn1** [comment]
@russel I probably wasn't clear enough, moving it to GNOME is not going to result in it being maintained. From what I see there is no team maintenance in GNOME, people work on what they want to work on. Tilix is not a core app and thus there is no incentive for GNOME to work on it, being in their gitlab doesn't magically fix things.

As a result moving the repo is just a change in scenery IMHO, it will continue to be unmaintained. Maybe there will be increased visibility but I don't think there will be any real change in that status. 

It's OK if Tilix goes away, software comes and then it goes. It's the circle of life in the open source world :)

**Comment by ximion** [comment]
Jup, moving it to the GNOME Gitlab doesn't mean people magically start to maintain it, the project may just as well die there.
But given that a lot of people do use Tilix, I can see people stepping up if they *already* have commit access to it anyway.
Tbh, I don't think Tilix actually needs a ton of maintenance given that there are active contributors to it - it just takes one to make releases somewhat frequently if there are changes and look over pull requests. That would mean the project would be in deep maintenance mode, but that's a different thing than dead.

**Comment by bilelmoussaoui** [comment]
> That would mean the project would be in deep maintenance mode, but that's a different thing than dead.

That's exactly what we are doing with FeedReader. We only bump the release number, include some very small fixes. It will probably work for Tilix, butt someone has to do that work.

Even if GNOME devs could have write access to the repository, I don't think it's a good idea to not have someone that "acts" as a maintainer and review/merge the MR instead of pushing to master directly.

**Comment by rikkimax** [comment]
Since this is actively used software, would you be interested in moving Tilix over to dlang-community?

This would help with maintenance (e.g. pulling PR's) but somebody will need to approve anything complex and act as a code owner. Basically they have to understand the code base and long term goals.

Some info on how we work: https://github.com/dlang-community/discussions

**Comment by 20after4** [comment]
I would be willing to help out with maintaining Tilix. I have a deep interest in terminals in general and I have spent a lot of time studying the historical and current state of the art in virtual terminals.   I've written a lot of terminal-based software and I have a fair understanding of VT100 escape sequences. Beyond terminals, I have fairly extensive free software development and release experience. My current day-job is Release Engineering at the Wikimedia Foundation.

I do not, however, have any experience with the D language, and only minimal C/C++ skills. 

I think @rikkimax's proposal sounds like a good way forward. Hopefully @gnunn1 agrees? If so then I'm willing and able to review PRs and I can help out with tests, builds & occasional releases.

**Comment by ximion** [comment]
@gnunn1 Since this just came up on Twitter, there are many people who do like Tilix and this bug will soon be a year old: If needed I could help out with Tilix maintenance, ideally together with others.
I do have D knowledge and maintain another project written in that language, I maintain Tilix for Debian/Ubuntu, and as such I already have an inherent interest to at least keep it compiling & working.
Some things I would like to do is to polish the Meson build support a bit (that's what we use in Linux distributions to build Tilix) and make a release of what is currently in master, as currently many changes and translation updates simply aren't getting to users due to the lack of releases.

As far as feature development is concerned, I couldn't do much though, as my FLOSS dev time is already occupied by work for Debian, Freedesktop and Purism.
If @20after4 is still interested in maintaining Tilix, I could probably help out with D related questions initially (and ideally he'd just pick up the language quickly, D is really not hard to learn if you already know some C/C++).
I also think that @rikkimax idea to move Tilix to a team-maintained place is good, and certainly better than the current state of it basically floating in limbo. Personally, I think for a project like this, GNOME World on https://gitlab.gnome.org/World would be a good place to have it, but dlang-community also has some things going for it (while GNOME has all the GTK+ knowledge, GNOME designers and deep knowledge about GTK+ as well as visibility within the GNOME community, dlang-community is where the D people are and a move within Github is easily done without migrating a lot of auxiliary data).

tl;dr I'd be willing to co-maintain Tilix, possibly help a future primary maintainer and keep it at least in basic maintenance mode so it works with new D language updates and GTK+ changes.

**Comment by gnunn1** [comment]
@ximion I've invited you as a collaborater to the repo

**Comment by fourstepper** [comment]
@ximion @gnunn1 @20after4 Sorry for the notification disturbance, but I was just wondering how things are looking between you guys? Is there any kind of conclusion to this? 

Thank you!

**Comment by ximion** [comment]
@fourstepper Unfortunately not... I got a bit overwhelmed with tasks (and the general hellscape that 2020 has been so far) and couldn't put the work into Tilix that it really needs. But even if I would have had more time I could only have done the bare minimum. Since I am subscribed to all activity around Tilix on GitHub (having push access), there's one thing I can say for sure: Tilix is a very popular project, and as such it gets a lot of attention in form of bug reports, feature requests and the occasional pull request. To handle this, it would really need one dedicated maintainer, or at least a team that can share that work - I alone couldn't currently spare the time with the other projects I am working on :-/
So tl;dr: We are - unless @gnunn1 figured something out that I don't know of yet - still looking for a maintainer.
In any case, don't let the D discourage you from taking this project - if you have some understanding of C/C++, picking up D isn't really that hard.

**Comment by ximion** [comment]
That being said, one thing I would *really* love to do is to make a new release and maybe include some of the proposed changes stalled in PRs - but making that release also will require a bit of testing and cleanup work, and isn't as simple as just pushing a button ;-)

**Comment by fourstepper** [comment]
Thank you for the answer :)

**Comment by RafalSkolasinski** [comment]
Tilix seems far more superior than any other terminals offered on Gnome (and beyond) due to its tiling functionality. Wonder if there would be even a slim chance convincing Gnome team on taking it on?

**Comment by stuaxo** [comment]
@RafalSkolasinski (I am saying this as a GNOME user): GNOME just doesn't seem focused on apps with lots of options and user customisability like Tilix so I'm not sure it would happen.

Having said that, they may be up for helping amplify the request for maintainers.

**Comment by ximion** [comment]
Well, getting Tilix into GNOME World ( https://gitlab.gnome.org/World ) and GNOME Circle ( https://circle.gnome.org/ ) would absolutely be a possibility, if @gnunn1 likes that option.
But that wouldn't be a magical fix for the lack of a proper maintainer, in fact I wouldn't be surprised if GNOME needed an active maintainer for the project to take care of it under the GNOME umbrella, rather than us dumping the project there and waiting for drive-by contributors (GNOME isn't Apache :-P).
The upside is that making use of Circle, Tilix would have access to some more "marketing" and have a louder voice to possibly attract users and developers.

We can do all this (and I can definitely help with the GNOME side), but I'm afraid we need a committed maintainer first (unfortunately I have so much stuff on my plate that I can't fill that role and only keep the code working, but not advance it substantially).

**Comment by RafalSkolasinski** [comment]
Hopefully. I strongly rely on it for my workflow, if not that I have no experience with dlang I'd be keen to do some work as well...

**Comment by ximion** [comment]
If you have some programming knowledge already, D has a nice mini-tutorial: https://tour.dlang.org/
Knowing some GTK is also useful though.

**Comment by RafalSkolasinski** [comment]
Mainly Python and Golang but unfortunately none for GTK :(

It'd be great to maybe at least gather together group of people interested in the project and organize some community call so we could synchronize on most pressing issues and prioritize some work?

**Comment by RafalSkolasinski** [comment]
I do have some experience in devops and automation so that's the area I could help in for example

**Comment by codewiz** [comment]
I switched from Tilix to Wezterm some time ago. It's harder to configure (no GUI, only a config file in Lua) but it's actively develped and has an impressive array of features: https://wezfurlong.org/wezterm/features.html

**Comment by RafalSkolasinski** [comment]
Thanks for the suggestion @codewiz, I've also seen people suggesting Kitty as a one to try. I am however very happy with Tilix and it does everything I need. I think the only issue I have is lack of native support for quake mode on Wayland.... And just being worried that it may stop being maintained

**Comment by alonsoweed** [comment]
Hello. 
I've been using Tilix for the last couple months and it rocks. I'm really sad to see it's struggling.
Maybe we could open a Patreon to donate and get a fixed amount to hire a developer or even to pay the creator a salary.

The project seems to have enough stars to get support!

**Comment by dejlek** [comment]
D is not the problem. Problem is the time needed to go through all the issues and PR for a project that is more and more popular...

**Comment by rene-s** [comment]
@gnunn1 Please feel free to add me as a maintainer for [tilix-web](https://github.com/gnunn1/tilix-web/pull/68), thank you very much. I would like to take care of the website.

I'm using Tilix every day and my workflow very much depends on Tilix and while I am not a confident D dev I'd like to at least update the website.

**Comment by okias** [comment]
I assume it doesn't have to be one person,
 - someone with light D knowledge for coordinating MRs? @20after4  (at least security updates)
 - @RafalSkolasinski for CI
 - @rene-s for the web

Tilix still works for me greatly, but would be nice to know, that at least it's kept in good shape.

**Comment by RafalSkolasinski** [comment]
We can use github actions - happy to help set up some pipelines.

The least I can do to make sure this great project continue to thrive - as it is my daily driver and it will probably remain so :)

**Comment by rene-s** [comment]
Well OK if someone is interesting in setting up a new website feel free to contact me.

**Comment by flyri0** [comment]
I can help with the website too

**Comment by jerome-antoine** [comment]
Hi guys ! 

May I ask what the current situation is about  the search for a new maintainer ? 

@RafalSkolasinski @rene-s  & @flyri0 : did you begin to coordinate with each other in any way ? 

I'd gladly participate in this project for reviving Tilix from ashes.

 And it'd also be cool to hear from @gnunn1 to know his position and get a real situation review. 

Please let me know if you have any info. 
I stay in touch 😉

**Comment by pavinjosdev** [comment]
@gnunn1 Thank you for this wonderful terminal app, I've been using Tilix for the past few weeks as my replacement for Gnome terminal and it's awesome. I wish more devs took this approach to software, i.e., once you've accomplished what you've set out to do and then some according to user feedback, it's enough to maintain the software and not add new features which invariably introduce more bugs.

**Comment by jerome-antoine** [comment]
Hi @gnunn !

Thank you for contacting me about the maintaining of Tilix terminal.

I currently use Tilix alonsige Foot and ATE (Awesome Terminal Emulator).

So do you still need my involvement as a co-maintainer ? And if so, do you
want me to do some particular works to begin ?

PS: congrats for your last release 1.9.6 !!

Cheers !!!

Jerome

Le mer. 15 nov. 2023, 07:30, Pavin Joseph ***@***.***> a
écrit :

**Comment by BarbzYHOOL** [comment]
@jerome-antoine it's not Gnunn1 who replied, but without his answer, I don't think anything can be done at the moment

-----------

I remember seeing a github repo where people could post an issue to have their project taken over by someone else. Cannot find it back sadly

**Comment by ximion** [comment]
I am currently doing the minimal maintenance work to keep Tilix compiling and fix the biggest bugs, but if someone wants to take over as maintainer proper and start feature development again, that would of course be amazing!

I can't add new people and grant them commit rights, or give anyone access to the website, but @jerome-antoine I could merge your PRs for sure. Do you have a bit of D experience? (it's not that hard, people who know basic C++ can probably pick it up fairly quickly)

I can also contact Gerald Nunn if we found a good maintainer going into the future to add new people properly and grant commit access. Some PRs would definitely help though :-) (Tilix' code is very readable, but with GTK4 there's probably a lot of porting work waiting in future)

**Comment by BarbzYHOOL** [comment]
thank you guys for your efforts, everybody is all about alacritty or kitty but Tilix is what I needed. Sry for offtopic

**Comment by gtirloni** [comment]
@gnunn1 thanks for your work on tilix, it's my terminal of choice.I absolutely can't live without the multiple panes and how easy tilix makes it to work with them. I've tried using more "modern" terminals with tmux and the experience is much worse, especially around scrolling and copy/paste operations.

I think supporting GTK4 would be a great first step into making tilix's maintenance more attractive. If there is interest from you @gnunn1 or someone else with D/GNOME knowledge, I'd be happy to discuss funding sources or to contribute meaningfully to a fundraiser.

**Comment by Obe1sance** [comment]
I recently switched to Wayland/Hyprland and while generally speaking Tilix seems to function as expected for me, I've recently had minimal issues. I stopped by here to see if it was still being maintained and saw this topic. 

I'm no programmer, but I do reverse engineer a bunch of stuff. I've never worked on a project before but after seeing this thing on what seems like passionate life-support, I want to help somehow. 

Even if only documentation, I'll fork it and figure out how to start contributing. This is hands down the most Elite teir Terminal Emulator for Aesthetics junkies, I have yet to find anything that even comes close. Literally every other terminal I've found hardly even has background image support, let alone built in color theming, and transparency effects. 

How tilix isn't the king of emulators absolutely eludes me. Please let me know how I can help. I'll learn whatever I have to do be useful. Cheers. 

P.S. Please don't let this project die. It's truely a work of art.

**Comment by ximion** [comment]
Oh, very nice! @dejlek would you still be interested in helping out / maintaining Tilix? I'll ping you on IRC soon (maybe tomorrow) :-)

**Comment by forgeOneHQ** [comment]
Do we any updates here? If it needs to be funded to give it to a new/additional maintainer I am open for it. Tilix my daily driver. Only Cosmic-Terminal is slowly getting where  tilix is what I see but I still prefer tilix.

**Comment by waldyrious** [comment]
As a fellow daily user of Tilix, I too would be willing to contribute a small monthly donation if that would help!

**Comment by 1rhino2** [comment]
> As many have probably noticed, activity on tilix has dwindled over the last year or so. From my perspective it does everything I need in a terminal emulator (well more actually) and I have little interest in adding features or even fixing bugs in features I don't use. I also want to spend more time on things more closely related to my day job and unfortunately that's not terminal emulators, GUIs, D or GNOME.
> 
> As a result I think some new blood with respect to Tilix would be beneficial. If you are interested in taking over the project let me know, ideally it would be someone who is already involved but we can work out a process for someone new.
> 
> Now does this mean tilix is completely dead if no one takes over? Probably not simply because I still use it everyday and will likely continue to put in the minimum effort to keep it working. However my motivation to push formal releases out the door versus just updating code in the repo will likely be pretty low.
> 
> With new energy I think there are some interesting directions Tilix could go so hopefully someone will be up to taking it on.

Hi @ximion and team,

I saw the call for maintainers in the README and wanted to offer a hand with active upkeep. 

**How I can help right away:**
* **Issue Triage:** Reviewing incoming reports, verifying bug reproductions, and helping answer user questions.
* **Code Reviews & PRs:** Reviewing pending pull requests and submitting targeted bug fixes.
* **Build & Dependencies:** Helping keep dependencies and GTK/VTE build setups clean and updated.

I'm ready to dive into the open issue queue and start submitting PRs to help lighten the load. Let me know where assistance is most needed!
